pkg/webhook/resources/upgrade/validator.go

```diff
@@ -307,6 +312,43 @@ func (v *upgradeValidator) checkMachines() error {
 	return nil
 }
 
+func (v *upgradeValidator) checkSingleReplicaVolumes() error {
+	// Upgrade should be rejected if any single-replica volume exists
+	nodes, err := v.nodes.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	// Skip for single-node cluster
+	if len(nodes) == 1 {
+		return nil
+	}
+
+	// Find all single-replica volumes
+	singleReplicaVolumes, err := v.lhVolumes.GetByIndex(indexeres.VolumeByReplicaCountIndex, "1")
+	if err != nil {
+		return err
+	}
+
+	volumeNames := make([]string, 0, len(singleReplicaVolumes))
+	for _, volume := range singleReplicaVolumes {
+		switch volume.Status.State {
+		case lhv1beta2.VolumeStateCreating, lhv1beta2.VolumeStateAttached, lhv1beta2.VolumeStateAttaching:
+			pvcNamespace := volume.Status.KubernetesStatus.Namespace
+			pvcName := volume.Status.KubernetesStatus.PVCName
+			volumeNames = append(volumeNames, pvcNamespace+"/"+pvcName)
+		}
+	}
+
+	if len(volumeNames) > 0 {
+		return werror.NewInternalError(
```

Comment:
Let's use NewInvalidError here, seems to make more sense here (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422)

Reply:
Updated, thanks!